File: src/java/org/javaturk/dp/ch05/flyweight/book/Page.java
package org.javaturk.dp.ch05.flyweight.book;

import java.util.ArrayList;
import java.util.List;

public class Page {
	private int no;
	private List<Line> lines;

	public Page(int no, int numberOfLine) {
		this.no = no;
		lines = new ArrayList<>(numberOfLine);
	}

	public void add(Line line) {
		lines.add(line);
	}

	public List<Line> getLines() {
		return lines;
	}

	public String toString() {
		StringBuilder sb = new StringBuilder();
		for (Line line : lines) {
			sb.append(line.toString());
		}
		return sb.toString();
	}
}

File: src/typescript/com/gokselkucuksahin/dp/structural-patterns/flyweight/book/page.ts
import {Line} from "./line";

export class Page {
  private no: number;
  private lines: Line[];

  constructor(no: number, numberOfLines: number) {
    this.no = no;
    this.lines = Array<Line>(numberOfLines);
  }

  add(line: Line): void {
    this.lines.push(line);
  }

  getLines(): Line[] {
    return this.lines;
  }

  toString(): string {
    return this.getLines().map(x => x.toString()).join('');
  }
}